### PR TITLE
[Core] Fix unpacking issue in unregister raw function

### PR DESCRIPTION
**Pull request by mk-armah**
# Description

**Fix unpacking error in unregister raw:**

**What** -  Resolved an issue in the `unregister_raw` method where an incorrect unpacking operation was causing a `ValueError ("not enough values to unpack (expected 2, got 1)")`. 

**Why** - This error occurred because the method was incorrectly expecting a tuple of entities and errors from the asyncio.gather call, but only receiving a single list due to the absence of zip unpacking. 

**How** -The fix involved using zip to correctly unpack the results into entities and errors, and then flattening these lists to handle multiple resource mappings properly. This change ensures that the method correctly processes the output of the asynchronous tasks and handles errors appropriately.

## Type of change

Please leave one option from the following and delete the rest:

- [ ] Bug fix (non-breaking change which fixes an issue)

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,14 @@ this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 <!-- towncrier release notes start -->
 
+## 0.9.8 (2024-08-01)
+
+
+### Bug Fixes
+
+- Fix unpacking error in unregister raw
```
Comment (matan84) [inline CHANGELOG.md:15]: Add more data here- what was broken, and how it was fixed

**Review by matan84 — CHANGES_REQUESTED** [review]
Good job, left one comment

**Review by yairsimantov20 — APPROVED** [review]
🌊 
Good job!

**Review by matan84 — APPROVED** [review]
LGTM

**Pull request merged**